source/robot_model/src/Model.cpp

```diff
@@ -299,23 +299,23 @@ Model::inverse_kinematics(const state_representation::CartesianPose& cartesian_p
   const std::chrono::nanoseconds dt(static_cast<int>(1e9));
   // initialization of the loop variables
   state_representation::JointPositions q(joint_positions);
-  state_representation::JointVelocities dq(this->get_robot_name(), this->get_number_of_joints());
-  state_representation::Jacobian J(this->get_robot_name(), this->robot_model_.nq, frame_name);
-  Eigen::MatrixXd J_b = J.data();
+  state_representation::JointVelocities dq(this->get_robot_name(), joint_positions.get_names());
+  state_representation::Jacobian J = this->compute_jacobian(joint_positions, frame_name);
```

Comment:
Why do you compute the jacobian here? It's gonna be recomputed before the first time its used

Reply:
right. I can just initialize it with all its parameters

Reply:
Ok I have changed it to just initialize the `Jacobian`. Because I needed the name of the base frame I also introduce a getter for it as I know need it in multiple places.